src/Circulator.sol

```diff
@@ -98,55 +93,56 @@ contract Circulator is ICirculator, FeeOperator, Pausable, EIP712, Nonces {
 
     /**
      * @notice Modifier to ensure that a given burn amount for a token doesn't exceed the allowed burn limit.
-     * @dev Queries the `burnLimitsPerMessage` from the `localMinter` to get the maximum allowed burn amount for the token.
+     * @dev Queries the `burnLimitsPerMessage` from the `tokenMinter` to get the maximum allowed burn amount for the token.
      * @param amount The amount of the token being requested to burn.
      */
     modifier onlyWithinBurnLimit(uint256 amount) {
-        uint256 _allowedBurnAmount = localMinter.burnLimitsPerMessage(circleAsset);
+        uint256 _allowedBurnAmount = tokenMinter.burnLimitsPerMessage(circleAsset);
         if (amount > _allowedBurnAmount) revert BurnAmountExceedsLimit();
         _;
     }
 
     /**
-     * @notice Teleport a specified amount to the destination domain and emits a `Circulate` event.
+     * @notice Circulate a specified amount to the destination domain and emits a `Circulate` event.
      * @dev This function burns a token amount for the given recipient and destination domain.
-     * @param _amount Amount to teleport.
+     * @param _inputAmount Amount to circulate.
      * @param _recipient The address of the recipient in bytes32 format.
      * @param _destinationDomain The ID of the destination domain.
-     * @return nonce Burn nonce for the teleport.
+     * @param _type Circulate type: Cctp or Across.
+     * @return nonce Burn nonce for the circulate.
      */
-    function circulate(uint256 _amount, bytes32 _recipient, uint32 _destinationDomain)
-        external
-        whenNotPaused
-        onlyWithinBurnLimit(_amount)
-        returns (uint64 nonce)
-    {
+    function circulate(
+        uint256 _inputAmount,
+        uint256 _outputAmount,
+        address _recipient,
+        uint32 _destinationDomain,
+        uint32 _fillDeadline,
+        CirculateType _type
+    ) external whenNotPaused onlyWithinBurnLimit(_inputAmount) returns (uint64 nonce) {
         // Check if fee is covered
-        uint256 fee = totalFee(_amount, _destinationDomain);
-        if (fee > _amount) revert FeeNotCovered();
+        uint256 fee = totalFee(_inputAmount, _destinationDomain);
+        if (fee > _inputAmount) revert FeeNotCovered();
 
-        uint256 burnAmt = _amount - fee;
+        uint256 burnAmount = _inputAmount - fee;
 
-        // Burn the token in TokenMessenger
-        IERC20(circleAsset).safeTransferFrom(msg.sender, address(this), _amount);
-        nonce = tokenMessenger.depositForBurn(burnAmt, _destinationDomain, _recipient, circleAsset);
+        nonce = _circulate(burnAmount, _outputAmount, _recipient, _destinationDomain, _fillDeadline, _type);
```

Comment:
why do we need this nonce? 
If we need it, we should make sure Across nonce is always different than CCTP nonce.
a possible solution could be using the first 4 bits for types. (nonce >> 4 << 4 + type mask)

Reply:
nonce is the depositId in Across and Cctp and will be used to track the order, they are both uint64. So it's possible to collide. But it's mainly for the API. So we can use mode + nonce to differentiate and track the deposit. :)